### ISSUE TITLE
Are separate types for IPv4 and IPv6 prefixes necessary?

### ISSUE THREAD

**Issue by grongor** [osrg/gobgp#2512]
Hello,

I was just going to fix some slice capacity issue and I started digging around ... and to be honest, I'm lost around the IPAddrPrefix, IPAddrPrefixDefault and IPv6AddrPrefix. It all seems kinda messy and unnecessary. Am I missing something? I think there should be only one "IPAddrPrefix" which handles both IPv4 and IPv6 ... if there isn't any reason for the distinction that I'm missing, I can take a look into merging these into one struct.

**Comment by fujita** [comment]
Yeah, I made several mistakes on these prefix stuff. But I guess that there are good number of users who use them. I don't think that it's a good idea to change them largely just because we can have the better design. It's too late.

**Comment by grongor** [comment]
Ultimately it's your decision of course, but I would like to point out that its expected to change/break things with the new major version, so I personally wouldn't be afraid to change things for the better with the new major version. It to me certainly feels like a better option than to just carry your "mistakes" alone forever.

**Comment by serejkus** [comment]
What do you think about creating an issue/wiki page that tracks some
possible breaking changes for future major releases?

сб, 25 дек. 2021 г., 16:25 Jakub Chábek ***@***.***>:
